app/src/main/java/com/keylesspalace/tusky/components/viewthread/ViewThreadFragment.kt

```diff
@@ -211,7 +212,7 @@ class ViewThreadFragment : SFragment(), OnRefreshListener, StatusActionListener,
                         threadProgressBar.cancel()
 
                         adapter.submitList(uiState.statusViewData) {
-                            if (viewModel.isInitialLoad) {
+                            if (lifecycle.currentState.isAtLeast(Lifecycle.State.STARTED) && viewModel.isInitialLoad) {
```

Comment:
Based on https://developer.android.com/topic/libraries/architecture/coroutines#restart I think `repeatOnLifecycle` is the thing to do here, around line 152:

```
        viewLifecycleOwner.lifecycleScope.launch {
            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) { // <-- new line
                viewModel.uiState.collect { uiState ->
```

Reply:
That should fix the crash potential for all the code here that operates on `binding`.

Reply:
I don't think so, the problem is that the `submitList` callback is called without regarding the lifecycle. The coroutine on the view lifecycle is already correctly cancelled, submitList just doesn't care and runs the callback anyway.